src/wasm-lib/kcl-to-core/src/conn_mock_core.rs

```diff
@@ -0,0 +1,467 @@
+                entity_id: _,
+                axis: _,
+                num_repetitions: _,
+                spacing: _,                                
+            } => {
+                // let num_transforms = transforms.len();
+                // let num_repetitions = transform.iter().map(|t| if t.replicate { 1 } else { 0 } ).sum();
+
+                // let mut base_code: String = format!(
+                //     r#"
+                //     std::vector<std::optional<Scene::Scene::LinearPatternTransform>> transforms_{cpp_id}({num_transforms});
+                // "#);
+                    
+                // for t in transform {
+                //     translations_xyz.push(t.translate.x.to_millimeters(state.units));
+                //     translations_xyz.push(t.translate.y.to_millimeters(state.units));
+                //     translations_xyz.push(t.translate.z.to_millimeters(state.units));
+                //     scale_xyz.push(t.scale.x);
+                //     scale_xyz.push(t.scale.y);
+                //     scale_xyz.push(t.scale.z);
+                // }
+    
+                // let entity_ids = generate_repl_uuids(*num_repetitions as usize);
+
+                // this_response = OkModelingCmdResponse::EntityLinearPattern {
+                //     data: kittycad::types::EntityLinearPattern {
+                //         entity_ids: entity_ids.clone(),
+                //     },
+                // };
+
+                // let mut base_code: String = format!(
+                //     r#"
+                //     auto reps_{cpp_id} = scene->entityCircularPattern(Utils::UUID("{}"), {num_repetitions}, glm::dvec3 {{ {}, {}, {} }}  * scaleFactor, glm::dvec3 {{ {}, {}, {} }}  * scaleFactor, {arc_degrees}, {rotate_duplicates});
+                // "#,
+                //     entity_id, axis.x, axis.y, axis.z, center.x, center.y, center.z
+                // );
+
+                // let repl_uuid_fix_code = codegen_cpp_repl_uuid_setters(&cpp_id, &entity_ids);
+                // base_code.push_str(&repl_uuid_fix_code);
+
+                // base_code
+                format!("//{:?}", cmd)
+            }
+            _ => {
+                //helps us follow along with the currently unhandled engine commands
+                format!("//{:?}", cmd)
+            }
+        };
+
+        ( new_code, this_response )
+    }
+}
+
+fn id_to_cpp(id: &uuid::Uuid) -> String {
+    let str = format!("{}", id);
+    str::replace(&str, "-", "_")
+}
+
+fn generate_repl_uuids(count: usize) -> Vec<uuid::Uuid> {
+    let mut repl_ids: Vec<uuid::Uuid> = Vec::new();
+
+    repl_ids.resize_with(count, uuid::Uuid::new_v4);
+    repl_ids
+}
+
+#[allow(clippy::needless_range_loop)]
+fn codegen_cpp_repl_uuid_setters(reps_id: &str, entity_ids: &[uuid::Uuid]) -> String {
+    let mut codegen = String::new();
+
+    for i in 0..entity_ids.len() {
+        let id = entity_ids[i];
+        let cpp_id = id_to_cpp(&entity_ids[i]);
```

Comment:
No need to disable clippy if you just want the index. The benefit of using iterators is that the compiler should be able to optimize away the bounds checking. Use `iter()` instead of `into_iter()` if you want to borrow values instead of move.
```suggestion
    for (i, id) in entity_ids.into_iter().enumerate() {
        let cpp_id = id_to_cpp(&id);
```